src/test/java/io/confluent/examples/streams/CustomStreamTableJoinIntegrationTest.java

```diff
@@ -0,0 +1,538 @@
+          streamBufferStore = (KeyValueStore<String, Pair<Double, Instant>>) context.getStateStore(streamBufferStoreName);
+          this.context = context;
+        }
+
+        /**
+         * A return value of `null` for this method represents a table tombstone record.  We need to discard such
+         * tombstone records in the downstream processing topology (see code further above in this file) because, in the
+         * context of this example of a custom stream-table join, they must not appear in the join output.  Note that we
+         * can't discard the tombstones directly in the ValueTransformerWithKey.
+         */
+        @Override
+        public Pair<Double, Long> transform(final String key, final Long value) {
+          LOG.info("Received table record ({}, {}) with timestamp {}", key, value, context.timestamp());
+          return possiblySendFullJoinRecord(key, value, context.timestamp());
+        }
+
+        private Pair<Double, Long> possiblySendFullJoinRecord(final String key,
+                                                              final Long tableValue,
+                                                              final long tableRecordTimestamp) {
+          if (tableValue != null) {
+            final Pair<Double, Instant> streamValue = streamBufferStore.get(key);
+            if (streamValue != null) {
+              // You can also incorporate timestamps of records into your join logic as shown here.
+              if (withinAcceptableBounds(streamValue.y, Instant.ofEpochMilli(tableRecordTimestamp))) {
+                LOG.info("Stream data available for key {}, sending fully populated join message ({}, ({}, {}))", key,
+                  key, streamValue.x, tableValue);
+                streamBufferStore.delete(key);
+                return new Pair<>(streamValue.x, tableValue);
+              } else {
+                LOG.info("Stream data available for key {} but not used because it is too old; sending tombstone", key);
+                return null;
+              }
+            } else {
+              LOG.info("Stream data not available for key {}; sending tombstone", key);
+              return null;
+            }
+          } else {
+            LOG.info("Table value for key {} is null (tombstone); sending tombstone", key);
+            return null;
+          }
+        }
+
+        private boolean withinAcceptableBounds(final Instant streamRecordTimestamp,
+                                               final Instant tableRecordTimestamp) {
+          return Duration.between(streamRecordTimestamp, tableRecordTimestamp)
+            .compareTo(approxMaxWaitTimePerRecordForTableData) <= 0;
+        }
+
+        @Override
+        public void close() {
+        }
+
+      };
+    }
+
+  }
+
+  private void writeInputDataToStream(final List<KeyValueWithTimestamp<String, Double>> inputStreamRecords)
+    throws ExecutionException, InterruptedException {
```

Comment:
I agree, but it's the auto-formatting rules we use, so I don't want to change it (because that will result in "fake" changes to this line in the future when auto-formatting is used).

Reply:
Update: I changed my code formatting settings accordingly, so it is better indented now.

(IntelliJ: Editor > Code Style > Java > Tabs and Indents: Continuation indent: 2 -> 4)